Add Figma documentation (#53)
primer/design: pages/design/tools/figma.mdx

import {Flex, Box, Text, BorderBox} from '@primer/components'

# Figma
Welcome to the Figma page of the Primer design system! Figma is used at GitHub as a single source of design, collaboration, and exploration.

<Text textAlign="center">
   <Flex alignItems='center' flexDirection='row' mt='4' mb='4'>
    <Box width={12/12}>
      <BorderBox height="250px" overflow='hidden'>
       <img src="https://user-images.githubusercontent.com/6846673/61730055-6ddf6000-ad2d-11e9-92ce-cd14fccc6851.png"/>
      </BorderBox>
    </Box>
   </Flex>
</Text>

> This page can help you learn how to use:

Reviewer: This page can help you become for familiar with

* [Team Library](#Library)
* [Styles](#Styles)
* [Components](#Components)
* [Prototyping and Wireframing](#Prototyping)
* [Collaboration](#Collaboration)

## <a name="Library"></a>Team Library

> GitHub has a Figma team that contains all of GitHub's design assets. In Figma, [styles](#Styles) and [components](#Components) can be reused across multiple files and projects with the [team library](https://help.figma.com/article/29-team-library) feature. With team libraries, you can enable and disable components styles within certain files. This makes it easy for designers to easily access Primer components from anywhere.

Reviewer: ```suggestion
GitHub has a Figma team that contains all of our design assets. In Figma, [styles](#Styles) and [components](#Components) can be reused across multiple files and projects with the [team library](https://help.figma.com/article/29-team-library) feature. With team libraries, you can enable and disable components styles within certain files. This makes it easy for designers to easily access Primer components from anywhere.
```

<Text textAlign="center" fontStyle='italic'>
   <Flex alignItems='center' flexDirection='row' mt='4' mb='4'>
    <Box width={12/12}>
      <BorderBox overflow='hidden'>
       <img width="100%" src="https://user-images.githubusercontent.com/6846673/61732332-3f17b880-ad32-11e9-8e18-bfbfc0267d81.gif"/>
      </BorderBox>
      <Text textAlign='center' color='gray.6'>Enable and disable libraries of components from across your team.</Text>
    </Box>
   </Flex>
</Text>

## <a name="Styles"></a>Styles

> Primer uses Figma styles to maintain a consistent look and feel across all of GitHub's UI. We have provided styles for color variables as well as our typography.

Reviewer: should link to the Figma styles resource here? "Primer uses [Figma styles](#).."

> Primer uses Figma styles to maintain a consistent look and feel across all of GitHub's UI. We have provided styles for color variables as well as our typography.

Reviewer: ```suggestion
Primer uses Figma styles to maintain a consistent look and feel across all of our UI. We have provided styles for color variables as well as our typography.
```

> Color styles are used to keep colors consistent across different products. Different type styles are used based on the importance and actionability of the text.

Reviewer: ```suggestion
```

I would remove this bit, it seems too generic to be useful atm. Think in the future we could flesh this out a bit more, but given the deadline I'd cut it.

## <a name="Components"></a>Components
Primer uses components for all of our common UI patterns. For components that have several different states, we use layers or groups within each component to quickly switch between states. For a full list of Primer Components, please see the [resources](#Resources) below.

<Text textAlign="center">
   <Flex alignItems='center' flexDirection='row' mt='4' mb='4'>
   <Box width={6/12} mr='4'>
     <BorderBox height="250px" overflow='hidden'>
      <img src="https://user-images.githubusercontent.com/6846673/61830583-5af98800-ae20-11e9-8aa9-459fc2d51133.gif"/>
     </BorderBox>
     <Text textAlign='center' color='gray.6'>Search for components in the assets menu and drag them onto the canvas.</Text>
   </Box>
    <Box width={6/12} mr='4'>
      <BorderBox height="250px" overflow='hidden'>
       <img src="https://user-images.githubusercontent.com/6846673/61830581-5af98800-ae20-11e9-93ab-9c33ff0dd2be.gif"/>
      </BorderBox>
      <Text textAlign='center' color='gray.6'>Some components have multiple states, which you can toggle in the layer list.</Text>
    </Box>
   </Flex>
</Text>

## <a name="Prototyping"></a>Prototyping and Wireframing

> We’ve developed a prototyping toolkit that includes a set of page templates and components to help you get started with exploration and prototyping in Figma. These pages are “medium” fidelity and have flexible elements that can be hidden or shown to work in different complexities. You can use the “Prototype” tab in the righthand sidebar to get started with interactive prototype projects. When you’re ready to present, simply hit the play (▶️) button in the top right.

Reviewer: ```suggestion
We’ve developed a prototyping toolkit that includes a set of page templates and components to help you get started with exploration and prototyping in Figma. These pages have flexible elements that can be hidden or shown to work in different fidelities. You can use the “Prototype” tab in the righthand sidebar to get started with interactive prototype projects. When you’re ready to present, simply hit the play (▶️) button in the top right.
```

<Text textAlign="center" fontStyle='italic'>
   <Flex alignItems='center' flexDirection='row' mt='4' mb='4'>
    <Box width={12/12}>
      <BorderBox overflow='hidden'>
       <img src="https://user-images.githubusercontent.com/6846673/61732575-bfd6b480-ad32-11e9-8e77-5a8c28b9bd88.png"/>
      </BorderBox>
    </Box>
   </Flex>
</Text>

Duplicate this [Figma file](https://www.figma.com/file/ex9hnSDJTr8fG24cTMyRJzHD/prototyping-pages?node-id=0%3A1) to your own project to use these templates for prototyping or exploration.

## <a name="Collaboration"></a>Collaboration

> All Hubbers have read-only access to all GitHub design files and edit access to files within their team. You can leave [Figma comments](https://help.figma.com/article/44-comments) to provide feedback within Figma and check [issues](https://help.github.com/en/articles/about-issues) to track decisions and progress. The design team works with incremental correctness, so share early and share often!

Reviewer: ```suggestion
All Hubbers have read-only access to all GitHub design files and edit access to files within their team. You can leave [Figma comments](https://help.figma.com/article/44-comments) to provide feedback and use [issues](https://help.github.com/en/articles/about-issues) to document design exploration, track decisions and progress. The design team values [incremental correctness](https://github.com/github/product-design/blob/master/principles-and-practices.md#incremental-correctness), so share early and share often.
```

## <a name="Resources"></a>Resources
Check out these places to learn more about Figma and GitHub:
- [Figma Design Documentation](https://help.figma.com/)
- [GitHub takes its collaborative culture to a new level](https://www.figma.com/blog/github-case-study/)
- [GitHub Primer Design](https://primer.style/design)

> List of current Primer Components:

Reviewer: We talked irl about this, but I think the list of components could come first here

import {Flex, Box, Text, BorderBox} from '@primer/components'

# Figma
Welcome to the Figma page of the Primer design system! Figma is used at GitHub as a single source of design, collaboration, and exploration. This page can help you learn how to use visual styles, library components, prototyping tools, and multiplayer collaboration.

<Text textAlign="center">
   <Flex alignItems='center' flexDirection='row' mt='4' mb='4'>
    <Box width={12/12}>
      <BorderBox height="250px" overflow='hidden'>
       <img src="https://user-images.githubusercontent.com/6846673/61730055-6ddf6000-ad2d-11e9-92ce-cd14fccc6851.png"/>
      </BorderBox>
    </Box>
   </Flex>
</Text>

## Styles

> GitHub’s Figma styles help give our products the same great feel across multiple products, sites, and places.

Reviewer 2: ```suggestion
Primer uses Figma styles to maintain a consistent look and feel across all of GitHub's UI. We have provided styles for color variables as well as our typography.
```

Color styles are used to keep colors consistent across different products. Different type styles are used based on the importance and actionability of the text.

<Text textAlign="center">
   <Flex alignItems='center' flexDirection='row' mt='4' mb='4'>
    <Box width={6/12} mr='4'>
      <BorderBox height="250px" overflow='hidden'>
       <img src="https://user-images.githubusercontent.com/6846673/61730793-ff9b9d00-ad2e-11e9-8f1b-dc52afce9bba.gif"/>
      </BorderBox>
      <Text textAlign='center' color='gray.6'>Each color has 10 variations, creating a spectrum of light and dark shades.</Text>
    </Box>
    <Box width={6/12} mr='4'>
      <BorderBox height="250px" overflow='hidden'>
       <img src="https://user-images.githubusercontent.com/6846673/61730796-01fdf700-ad2f-11e9-9208-065eabc8e806.gif"/>
      </BorderBox>
      <Text textAlign='center' color='gray.6'>The type system includes bold, uppercase, and small variations for body and header text.</Text>
    </Box>
   </Flex>
</Text>

Check out our [color guidelines](https://primer.style/design/foundation/color) and [typography guidelines](https://primer.style/design/foundation/typography) for more details.

## Components

> Components are elements that have a common visual style and purpose across GitHub products. While they may share the same basic design, some have hidden layers or groups for different states or colors. In Figma, you can access these components from the “Assets” tab at the top of the lefthand sidebar. Here's a list of our current components (and their Figma files):

Reviewer 2: ```suggestion
Primer uses components for all of our common UI patterns. For components that have several different states, we use layers or groups within each component to quickly switch between states.

In Figma, you can access these components from the “Assets” tab at the top of the lefthand sidebar. Here's a list of our current components (and their Figma files):
```

- [Tooltips](https://www.figma.com/file/EmivQqEsAexN5eEclS78JTVH/primer-tooltips)

## Prototyping and Wireframing

> We’ve developed a set of page templates to help you get started with exploration and prototyping in Figma. These pages are “medium” fidelity and have flexible elements that can be hidden or shown to work in different complexities. You can use the “Prototype” tab in the righthand sidebar to get started with interactive prototype projects. When you’re ready to present, simply hit the play button in the top right.

Reviewer 2: ```suggestion
We’ve developed a prototyping toolkit that includes a set of page templates and components to help you get started with exploration and prototyping in Figma. These pages are “medium” fidelity and have flexible elements that can be hidden or shown to work in different complexities. You can use the “Prototype” tab in the righthand sidebar to get started with interactive prototype projects. When you’re ready to present, simply hit the play (▶️) button in the top right.
```
